[Sparse] Support strided tensor when calling old DGL APIs

czkkkkkk opened the pull request:
## Description
Resolve #5502.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes
<!-- You could use following template
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)
-->

jermainewang commented:
1. Double check all other operators.
2. What is the best practice moving forward? Shall we have a safe guard utility to check/handle strided cases?
3. Add proper unit tests. I'm curious if ChatGPT can help us convert our UTs automatically.

frozenbugs commented:
+1 for unit test, for all operators, we should have a unit test for strided cases. Is there any other cases / types of tensor.

czkkkkkk commented:
> * Double check all other operators.
> * What is the best practice moving forward? Shall we have a safe guard utility to check/handle strided cases?

I added `.contiguous` when converting a Pytorch tensor to a DGL array. Then we don't need to handle strided tensors in any operator.

> * Add proper unit tests. I'm curious if ChatGPT can help us convert our UTs automatically.
Added. The github copilot provides useful hints when writing tests.

czkkkkkk commented:
> Is there any other cases / types of tensor.

I don't think so. Stride is a very special case.

jermainewang commented on tests/python/pytorch/sparse/test_matmul.py:
```diff
@@ -30,6 +31,7 @@ def test_spmm(create_func, shape, nnz, out_dim):
     else:
         X = torch.randn(shape[1], requires_grad=True, device=dev)
 
+    X = rand_stride(X)
```
This means we are only testing strided tensor while not testing normal tensors, though I feel it is a minor issue.

czkkkkkk replied:
Actually the stride is a random number in range [1, 4]. It is possible that the stride is one.

jermainewang replied:
Oh, we should avoid randomness in unit tests as much as possible. Otherwise, it is possible that some developers got lucky but later others fail the CI pipeline due to non-related errors. Please fix this.

frozenbugs commented on tests/python/pytorch/sparse/utils.py:
```diff
@@ -1,7 +1,7 @@
 import torch
 
-from dgl.sparse import from_coo, from_csc, from_csr, SparseMatrix
+from dgl.sparse import from_coo, from_csc, from_csr, SparseMatrix, spmatrix
```
remove from_coo

jermainewang reviewed (changes requested):
Just realized that the stride is a random int from 1 to 4. This is bad practice because stride=1 means no stride. Please fix the number in UT.

czkkkkkk commented:
> Just realized that the stride is a random int from 1 to 4. This is bad practice because stride=1 means no stride. Please fix the number in UT.

Done
